lib/src/module/built_in.dart

```diff
@@ -39,8 +41,16 @@ class BuiltInModule<T extends AsyncCallable> implements Module<T> {
           : UnmodifiableMapView(
               {for (var callable in callables) callable.name: callable}));
 
+  /// Returns a map from [variables]' names to their values.
+  static Map<String, V> _variableMap<V>(Map<String, V> variables) =>
+      UnmodifiableMapView(
+          variables == null ? {} : UnmodifiableMapView(variables));
+
   void setVariable(String name, Value value, AstNode nodeWithSpan) {
-    throw SassScriptException("Undefined variable.");
+    if (!variables.containsKey(name)) {
+      throw SassScriptException("Undefined variable.");
+    }
+    throw SassScriptException("Cannot modify built-in variable \$$name.");
```

Comment:
You don't need to include the variable name here, because it'll always be present in the error's source span (that's why we don't mention it above).

Reply:
Makes sense, fixed.